### PR TITLE
Update for new snmalloc.

### PR DESCRIPTION
**Pull request by davidchisnall**
This is currently using the microsoft/snmalloc#501 in the submodule.  It
will track that until merged.

### CODE REVIEW

File: experiments/process_sandbox/include/process_sandbox/sandbox.h [hunk 1]
```diff
@@ -288,8 +202,109 @@ namespace sandbox
...
+       */
+      LocalState(void* start, size_t size);
+
+      /**
+       * Predicate to test whether an object of size `sz` starting at `ptr`
+       * is within the region managed by this memory provider.
+       */
+      bool contains(const void* ptr, size_t sz)
+      {
+        return (ptr >= base) && (pointer_diff(ptr, (void*)top) >= sz);
```
Comment (ihaller) [inline experiments/process_sandbox/include/process_sandbox/sandbox.h:291]: Small nitpick, but isn't `top` already a `void*`?

File: src/rt/test/logging.h [hunk 2]
```diff
@@ -327,7 +327,7 @@ namespace Logging
 
     static void dump_flight_recorder(std::string id = "")
     {
-      static std::atomic_flag dump_in_progress = ATOMIC_FLAG_INIT;
+      static snmalloc::FlagWord dump_in_progress = ATOMIC_FLAG_INIT;
```
Comment (mjp41) [inline src/rt/test/logging.h:330]:
```suggestion
      static snmalloc::FlagWord dump_in_progress;
```

**Review by mjp41 — APPROVED** [review]
LGTM, one minor nit.

**Comment by ihaller** [comment]
LGTM too.

**Comment by davidchisnall** [comment]
@marioskogias, is `func-con-external_event_source` known to be flaky?  I don't think anything here has changed anything connected to it.

**Comment by davidchisnall** [comment]
In that case, am I okay to merge with the failing test?

**Pull request merged**